src/controllers/MLFacetQueryController.ts

```diff
@@ -5,7 +5,11 @@ import { IFacetRequest, IFacetRequestValue } from '../rest/Facet/FacetRequest';
 import { FacetSortCriteria } from '../rest/Facet/FacetSortCriteria';
 
 export class MLFacetQueryController {
-  constructor(public facet: MLFacet) {}
+  public numberOfValuesToRequest: number;
```

Comment:
If we encapsulate the logic to change this value as per my comment below, this could become a `private` property